autoload/codefmt/buildifier.vim

```diff
@@ -0,0 +1,74 @@
+" Copyright 2017 Google Inc. All rights reserved.
+"
+" Licensed under the Apache License, Version 2.0 (the "License");
+" you may not use this file except in compliance with the License.
+" You may obtain a copy of the License at
+"
+"     http://www.apache.org/licenses/LICENSE-2.0
+"
+" Unless required by applicable law or agreed to in writing, software
+" distributed under the License is distributed on an "AS IS" BASIS,
+" WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+" See the License for the specific language governing permissions and
+" limitations under the License.
+
+
+let s:plugin = maktaba#plugin#Get('codefmt')
+let s:registry = s:plugin.GetExtensionRegistry()
+
+
+""
+" @private
+"
+" Formatter provider for Bazel BUILD files using buildifier.
+function! codefmt#buildifier#GetFormatter() abort
+  let l:formatter = {
+      \ 'name': 'buildifier',
+      \ 'setup_instructions': 'Install buildifier' .
+          \ '(https://github.com/bazelbuild/buildifier).'}
+
+  function l:formatter.IsAvailable() abort
+    return executable(s:plugin.Flag('buildifier_executable'))
+  endfunction
+
+  function l:formatter.AppliesToBuffer() abort
+    return &filetype is# 'bzl'
+  endfunction
+
+  ""
+  " Reformat the current buffer with buildifier or the binary named in
+  " @flag(buildifier)
+  " @throws ShellError
+  function l:formatter.Format() abort
```

Comment:
Can you also request a range formatting option at https://github.com/bazelbuild/buildifier/issues? I like to make sure all our supported formatters are aware it would help our integration work more smoothly. =)